be/src/Project/Web/Seller/Seller.Web/Areas/Products/ModelBuilders/DuplicateCategoryFormModelBuilder.cs

```diff
@@ -0,0 +1,118 @@
+﻿using Foundation.Extensions.ModelBuilders;
+using Foundation.Localization;
+using Foundation.Media.Services.MediaServices;
+using Foundation.PageContent.Components.ListItems.ViewModels;
+using Microsoft.AspNetCore.Routing;
+using Microsoft.Extensions.Localization;
+using Seller.Web.Areas.Products.ComponentModels;
+using Seller.Web.Areas.Products.DomainModels;
+using Seller.Web.Areas.Products.Repositories;
+using Seller.Web.Areas.Products.ViewModels;
+using Seller.Web.Areas.Shared.Repositories.Media;
+using Seller.Web.Shared.Definitions;
+using Seller.Web.Shared.ViewModels;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace Seller.Web.Areas.ModelBuilders.Products
+{
+    public class DuplicateCategoryFormModelBuilder : IAsyncComponentModelBuilder<DuplicateCategoryComponentModel, CategoryFormViewModel>
+    {
+        private readonly ICategoriesRepository categoriesRepository;
+        private readonly IMediaItemsRepository mediaItemsRepository;
+        private readonly IStringLocalizer globalLocalizer;
+        private readonly IStringLocalizer productLocalizer;
+        private readonly IMediaService mediaService;
+        private readonly LinkGenerator linkGenerator;
+
+        public DuplicateCategoryFormModelBuilder(
+            ICategoriesRepository categoriesRepository,
+            IMediaItemsRepository mediaItemsRepository,
+            IStringLocalizer<GlobalResources> globalLocalizer,
+            IStringLocalizer<ProductResources> productLocalizer,
+            IMediaService mediaService,
+            LinkGenerator linkGenerator)
+        {
+            this.categoriesRepository = categoriesRepository;
+            this.mediaItemsRepository = mediaItemsRepository;
+            this.globalLocalizer = globalLocalizer;
+            this.productLocalizer = productLocalizer;
+            this.mediaService = mediaService;
+            this.linkGenerator = linkGenerator;
+        }
+
+        public async Task<CategoryFormViewModel> BuildModelAsync(DuplicateCategoryComponentModel componentModel)
+        {
+            var viewModel = new CategoryFormViewModel
```

Comment:
**suggestion:** let's move code snippets common for DuplicateCategoryFormModelBuilder and CategoryFormModelBuilder to a common CategoryBaseFormModelBuilder :) It would be cool to do the same for duplicating products (in a separate PR) so that we don't duplicate code and don't have to maintain it in multiple places.

Reply:
Okay :)